src/world/entity.rs

```diff
@@ -331,6 +346,39 @@ impl<'a> Join for &'a EntitiesRes {
 
 unsafe impl<'a> ParJoin for &'a EntitiesRes {}
 
+/// An entity builder from `EntitiesRes`.  Allows building an entity with its
+/// components if you have mutable access to the component storages.
+pub struct EntityResBuilder<'a> {
+    /// The entity being built
+    pub entity: Entity,
+    /// The active borrow to `EntitiesRes`, used to delete the entity if the
+    /// builder is dropped without called `build()`.
+    pub entities: &'a EntitiesRes,
```

Comment:
I am also not sure why making this public would be useful, but there is no harm in this case... Well there could be if you had multiple worlds and started mixing their types, but that's rare.